tensorflow_quantum/python/optimizers/rotosolve_minimizer_test.py

```diff
@@ -0,0 +1,150 @@
+    @tf.function
+    def func(params):
+        """A function that can be used by tfq.optimizer.rotosolve_minimize.
+
+        Args:
+           params [in]: a 1D tf.Tensor.
+
+        Returns:
+            Loss function value
+        """
+
+        # update the parameters of the model
+        start = 0
+        for i, size in enumerate(sizes):
+            model.trainable_variables[i].assign(
+                tf.reshape(params[start:start + size], shape))
+            start += size
+
+        # evaluate the loss
+        loss_value = loss(model(train_x, training=True), train_y)
+        return loss_value
+
+    return func
+
+
+class RotosolveMinimizerTest(tf.test.TestCase, parameterized.TestCase):
+    """Tests for the rotosolve optimization algorithm."""
+
+    def test_function_optimization(self):
+        """Optimize a simple sinusoid function."""
+
+        n = 10  # Number of parameters to be optimized
+        coefficient = tf.random.uniform(shape=[n])
+        min_value = -tf.sum(tf.abs(coefficient))
+
+        func = lambda x: tf.sum(tf.sin(x) * coefficient)
+
+        result = rotosolve_minimizer.minimize(func, np.random.random(n))
+
+        self.assertAlmostEqual(func(result['position']), min_value)
+        self.assertAlmostEqual(result['objective_value'], min_value)
+        self.assertTrue(result['converged'])
+
+    def test_keras_model_optimization(self):
+        """Optimizate a PQC based keras model."""
+
+        x = np.asarray([
+            [0, 0],
+            [0, 1],
+            [1, 0],
+            [1, 1],
+        ], dtype=float)
+
+        y = np.asarray([[-1], [1], [1], [-1]], dtype=float)
+
+        def convert_to_circuit(input_data):
+            """Encode into quantum datapoint."""
+            values = np.ndarray.flatten(input_data)
+            qubits = cirq.GridQubit.rect(1, 2)
+            circuit = cirq.Circuit()
+            for i, value in enumerate(values):
+                if value:
+                    circuit.append(cirq.X(qubits[i]))
+            return circuit
+
+        x_circ = util.convert_to_tensor([convert_to_circuit(x) for x in x])
+
+        # Create two qubits
+        q0, q1 = cirq.GridQubit.rect(1, 2)
+
+        # Create an anzatz on these qubits.
+        a, b = sympy.symbols('a b')  # parameters for the circuit
+        circuit = cirq.Circuit(
+            cirq.rx(a).on(q0),
+            cirq.ry(b).on(q1), cirq.CNOT(control=q0, target=q1))
+
+        # Build the Keras model.
+        model = tf.keras.Sequential([
+            # The input is the data-circuit, encoded as a tf.string
+            tf.keras.layers.Input(shape=(), dtype=tf.string),
+            # The PQC layer returns the expected value of the
+            # readout gate, range [-1,1].
+            pqc.PQC(circuit, cirq.Z(q1)),
+        ])
+
+        def hinge_loss(y_true, y_pred):
+            # Here we use hinge loss as the cost function
+            return tf.reduce_mean(tf.cast(1 - y_true * y_pred, tf.float32))

```

Comment:
Why did we opt to not use the built in tf hinge loss ?

Reply:
There is no particular reason. Swapped to the built-in hinge loss function in 4053d7442db733859fdbbe8fa0e9ccfaa6ce1d09